src/views/Characters/Characters.js

```diff
@@ -0,0 +1,42 @@
+
+import React, { useEffect, useState } from 'react';
+import { fetchCharacters } from '../../services/characters';
+import Filter from '../../components/Controls/Filter';
+
+export default function Characters() {
+  const [characters, setCharacters] = useState([]);
+  const [error, setError] = useState('');
+  const [races, setRaces] = useState('All');
+
+  const options = ['All', 'Dwarf', 'Elf', 'Hobbit', 'Human', 'Maiar', 'Orc'];
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const characterData = await fetchCharacters('All');
+        setCharacters(characterData);
+      } catch (e) {
+        setError(e.message);
+      }
+    };
+    fetchData();
+  }, []);
+
+  const filterCharacters = () => {
+    const filtered = characters.filter((character) => character.race === races || races === 'All');
+    return filtered;
+  };
```

Comment:
Nice work filtering characters with a function here. We can also pass in our 'races' state to the fetchCharacters function to filter the characters that way.